### PR TITLE
Implement `getMetadata` for active JS scripts

### PR DESCRIPTION
**Pull request by ricekot**
Part of #440.

Also add relevant alert tags and assign categories. The scan rule IDs for these scripts were taken from the [scanners.md](https://github.com/zaproxy/zaproxy/blob/aedad6818c562a741115333a660d5de09f1780d9/docs/scanners.md) file.

The status of these scripts has been set to `alpha` since community-scripts is an alpha add-on.

### CODE REVIEW

**Comment by kingthorin** [comment]
FYI I haven't forgotten this. I did go through a number and they seemed fine. I'll try to finish a review in the next few days.

**Comment by psiinon** [comment]
image

Looks pretty good 😁 
Can we add meta data so we can link to the script source code? This might require code changes of course.
If it does then maybe also change to be able to report the alert type?
It would be better if it was "Alert Script" in this case

**Comment by kingthorin** [comment]
I finally got to read through all of them, seems fine to me. Are there further changes/tweaks coming or shall I go ahead and approve?

**Review by kingthorin — APPROVED** [review]
Thanks!

File: active/Cross Site WebSocket Hijacking.js [hunk 1]
```diff
@@ -41,31 +41,44 @@ var Base64 = Java.type("java.util.Base64");
...
+  - https://tools.ietf.org/html/rfc6455#section-10.2
+category: server
+risk: high
+confidence: medium
+cweId: 346  # CWE-346: Origin Validation Error, http://cwe.mitre.org/data/definitions/346.html
+wascId: 9  # WASC-9 Cross Site Request Forgery, http://projects.webappsec.org/w/page/13246919/Cross%20Site%20Request%20Forgery
+alertTags:
+  OWASP_2021_A01: https://owasp.org/Top10/A01_2021-Broken_Access_Control/
+  OWASP_2017_A05: https://owasp.org/www-project-top-ten/2017/A5_2017-Broken_Access_Control.html
+  WSTG-v42-CLNT-10: https://owasp.org/www-project-web-security-testing-guide/v42/4-Web_Application_Security_Testing/11-Client-side_Testing/10-Testing_WebSockets
```
Comment (thc202) [inline active/Cross Site WebSocket Hijacking.js:68]: We should look at a way of using the common lib for these as well.

File: active/JWT None Exploit.js [hunk 2]
```diff
@@ -4,18 +4,35 @@ var Cookie = Java.type("java.net.HttpCookie");
...
+  which bypasses the JWT hash verification.
+solution: >
+  Use a secure JWT library, and (if your library supports it) restrict the allowed hash algorithms.
+references:
+  - https://www.sjoerdlangkemper.nl/2016/09/28/attacking-jwt-authentication/
+category: server
+risk: high
+confidence: medium
+cweId: 347
+wascId: 15
```
Comment (thc202) [inline active/JWT None Exploit.js:26]: Comments were dropped but in the other script they were kept (don't mind either way just saying for consistency).

File: active/JWT None Exploit.js [hunk 3]
```diff
@@ -128,20 +145,9 @@ function scanNode(as, msg) {
...
-    SOLUTION,
-    "Cookie: " + cookie.getName() + "=" + payload,
-    REFERENCE,
-    CWEID,
-    WASCID,
-    msg
-  );
+  as.newAlert()
+    .setEvidence("Cookie: " + cookie.getName() + "=" + payload)
+    .setUri(msg.getRequestHeader().getURI().toString())
```
Comment (thc202) [inline active/JWT None Exploit.js:150]: Can be removed, it's the default from the message set.

File: active/gof_lite.js [hunk 4]
```diff
@@ -139,41 +145,16 @@ function mutate(msg, mutationString) {
 
 function raiseAlert(wasSuccess, msg, origMsg, as) {
   var url = msg.getRequestHeader().getURI().toString();
```
Comment (thc202) [inline active/gof_lite.js:147]: The url could be dropped.

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Comment by thc202** [comment]
@psiinon do you want to check this again?

**Review by psiinon — APPROVED** [review]

**Pull request merged**